### PR TITLE
get current location for overview map center, put a marker there

### PR DESCRIPTION
**Pull request by stevecassidy**
# Get current location for overview map center

## JIRA Ticket

None

## Description

The overview map had a fixed center point somewhere in Africa.  More useful to move it
to the current location.

## Proposed Changes

- get the current location when displaying the overview map, use it as the center point 
- add a marker (+) at the current location on the map

## How to Test

Open a notebook with geo-locations (BSS) and view the Map. See the cross at your current location.
Move the map, click the circle 'center map' button and see it center the map on your location.

## Checklist

- [x] I have confirmed all commits have been signed.
- [x] I have added JSDoc style comments to any new functions or classes.
- [x] Relevant documentation such as READMEs, guides, and class comments are updated.

### CODE REVIEW

**Comment by luke-mcfarlane-rocketlab** [comment]
Have a strange issue where it worked once but doesn't seem to work every time

File: app/src/gui/components/notebook/overview_map.tsx [hunk 1]
```diff
@@ -139,44 +140,94 @@ export const OverviewMap = (props: OverviewMapProps) => {
...
+      }),
     });
 
-    return theMap;
-  }, []);
+    const centerFeature = {
+      type: 'Feature',
+      geometry: {
+        type: 'Point',
+        coordinates: map_center,
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/overview_map.tsx:219]: is this safe if data is undefined - what's the behaviour?

File: app/src/gui/components/notebook/overview_map.tsx [hunk 2]
```diff
@@ -139,44 +140,94 @@ export const OverviewMap = (props: OverviewMapProps) => {
   const mapRef = useRef<Map | undefined>();
   mapRef.current = map;
 
-  const map_center = [30, -10];
+  const {data: map_center} = useQuery<Array<number>>({
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/overview_map.tsx:143]: nitpick: react-query course advises type annotating, if at all, as close to the data fetch as possible. I believe the generics on useQuery can muddle things up - so can just remove the Array<number> here, or if needed annotate line 147.

Comment (PeterBaker0) [reply app/src/gui/components/notebook/overview_map.tsx:143]: I know I've done this too

**Review by PeterBaker0 — COMMENTED** [review]
Haven't tested, just spying code.

File: app/src/gui/components/notebook/overview_map.tsx (hunk 2 above)
Comment (stevecassidy) [reply app/src/gui/components/notebook/overview_map.tsx:143]: Moved the types to the queryFn

File: app/src/gui/components/notebook/overview_map.tsx (hunk 1 above)
Comment (stevecassidy) [reply app/src/gui/components/notebook/overview_map.tsx]: Have put a guard around this.

**Review by PeterBaker0 — APPROVED** [review]
Works well after discussed changes re: race conditions. 

Would recommend some refactoring when we visit it again to do with error conditions and not relying on react query isLoading to check if something is happening 'for the first time' - would prefer refs to track this if needed.

**Pull request merged**